session/store.js

```diff
@@ -47,12 +47,27 @@ util.inherits(Store, EventEmitter)
  * @api public
  */
 
-Store.prototype.regenerate = function(req, fn){
+Store.prototype.regenerate = function(req, options, fn){
   var self = this;
-  this.destroy(req.sessionID, function(err){
+  if (options instanceof Function) {
+    fn = options;
+    options = {};
+  }
+  options = options || {};
+  var destroy = options.destroy;
+  if (destroy === undefined) {
+    destroy = true;
+  }
+
+  if ( destroy ) {
+    this.destroy(req.sessionID, function(err){
+      self.generate(req);
+      fn(err);
+    });
+  } else {
     self.generate(req);
-    fn(err);
-  });
+    fn(null);
```

Comment:
Please make sure you don't "release Zalgo" by making an API sometimes sync, and sometimes async. See http://blog.izs.me/post/59142742143/designing-apis-for-asynchrony for a complete explanation. The API should be always one or the other.